zebra-chain/src/sprout/tests/tree.rs

```diff
@@ -0,0 +1,71 @@
+//! Tests for Sprout Note Commitment Trees.
+
+use hex::FromHex;
+
+use crate::sprout::commitment::NoteCommitment;
+use crate::sprout::tests::test_vectors;
+use crate::sprout::tree;
+
+/// Tests if empty roots are generated correctly.
+#[test]
+fn empty_roots() {
+    zebra_test::init();
+
+    for i in 0..tree::EMPTY_ROOTS.len() {
+        assert_eq!(
+            hex::encode(tree::EMPTY_ROOTS[i]),
+            // The test vector is in reversed order.
+            test_vectors::HEX_EMPTY_ROOTS[tree::MERKLE_DEPTH - i]
+        );
+    }
+}
+
+/// Tests if we have the right unused (empty) leaves.
+#[test]
+fn empty_leaf() {
+    assert_eq!(
+        tree::NoteCommitmentTree::uncommitted(),
+        test_vectors::EMPTY_LEAF
+    );
+}
+
+/// Tests if we can build the tree correctly.
+#[test]
+fn incremental_roots() {
+    zebra_test::init();
+
+    let mut leaves = vec![];
+
+    let mut incremental_tree = tree::NoteCommitmentTree::default();
+
+    for (i, cm) in test_vectors::COMMITMENTS.iter().enumerate() {
+        let bytes = <[u8; 32]>::from_hex(cm).unwrap();
+        let cm = NoteCommitment::from(bytes);
+
+        // Test if we can append a new note commitment to the tree.
+        let _ = incremental_tree.append(cm);
+        let incremental_root_hash = incremental_tree.hash();
+        assert_eq!(hex::encode(incremental_root_hash), test_vectors::ROOTS[i]);
+
+        // The root hashes should match.
+        let incremental_root = incremental_tree.root();
+        assert_eq!(incremental_root.0, incremental_root_hash);
```

Comment:
Try using `From<Root> for [u8; 32]` vs relying on the inner type being public:

```suggestion
        assert_eq!(incremental_root.into(), incremental_root_hash);
```